src/plugins/Screenrecord.js

```diff
@@ -0,0 +1,233 @@
+        this.mediaRecorder = null;
+        this.isRecording = false;
+        this.startTime = 0;
+        this.displayInterval = null;
+        this.timerElement = null;
+    }
+
+    /**
+     * Add the buttons to the renderer toolbar.
+     */
+    registerToolbarButton() {
+        this.toolbarBtn = this.instance.toolbarManager.registerButton({
+            id: this.constructor.name,
+            iconClass: 'gm-screenrecord-button',
+            title: this.i18n.SCREENRECORD_TITLE || 'Screenrecord',
+            onClick: () => {
+                this.onScreenrecordClick();
+            },
+        });
+    }
+
+    /**
+     * MediaRecorder recorded data available callback.
+     *
+     * @param {Event} event Event.
+     */
+    handleDataAvailable(event) {
+        if (event.data && event.data.size > 0) {
+            this.recordedBlobs.push(event.data);
+        }
+    }
+
+    /**
+     * Get file extension for a given MIME type.
+     *
+     * @param  {string} mime MIME type.
+     * @return {string}      File extention
+     */
+    getExtensionFromMimeType(mime) {
+        const types = {
+            'video/x-msvideo': '.avi',
+            'video/mpeg': '.mpeg',
+            'video/ogg': '.ogv',
+            'video/webm': '.webm',
+            'video/3gpp': '.3gp',
+            'video/3gpp2': '.3g2',
+            'video/mp4': '.mp4',
+            'video/x-matroska': '.mkv',
+            'video/x-flv': '.f4v',
+        };
+
+        let extension = types[mime.toLowerCase()];
+        if (!extension) {
+            log.warn('Unknown MIME type: ', mime);
+            extension = '.webm';
+        }
+        return extension;
+    }
+
+    /**
+     * Download recorded screenrecord.
+     */
+    downloadScreenrecord() {
+        const blob = new Blob(this.recordedBlobs, {type: this.mediaRecorder.mimeType});
+        const url = window.URL.createObjectURL(blob);
+        const a = document.createElement('a');
+        a.style.display = 'none';
+        a.href = url;
+        a.download = 'device' + this.getExtensionFromMimeType(this.mediaRecorder.mimeType);
+        document.body.appendChild(a);
+        a.click();
+        setTimeout(() => {
+            document.body.removeChild(a);
+            window.URL.revokeObjectURL(url);
+        }, 100);
+        this.isRecording = false;
+    }
+
+    /**
+     * Update UI based on MediaRecorder availability.
+     */
+    checkForMediarecorder() {
+        if (typeof MediaRecorder === 'undefined') {
+            this.toolbarBtn.disable();
+        }
+    }
+
+    /**
+     * Display screenrecord timer indicator.
+     */
+    displayTimer() {
+        const endTime = new Date();
+
+        // compute seconds
+        let timeDiff = endTime - this.startTime;
+        timeDiff /= 1000;
+        const seconds = Math.round(timeDiff % 60);
+
+        // compute minutes
+        timeDiff = Math.floor(timeDiff / 60);
+        const minutes = Math.round(timeDiff % 60);
```

Comment:
```suggestion
        const minutes = Math.floor(timeDiff / 60);
```

Since we're comparing `MAX_SCREENRECORD_LENGTH_IN_MINUTES` to the duration in minutes, the modulo would provide incorrect results if  `MAX_SCREENRECORD_LENGTH_IN_MINUTES` were bigger than one hour (it's not the case today, so no bug for now).

Example, let's say the elapsed time is 63 minutes and 27 seconds:
endTime - startTime = 3807000.

```javascript
        let timeDiff = endTime - this.startTime; // 3807000
        timeDiff /= 1000; // 3807
        const seconds = Math.round(timeDiff % 60); //27

        // compute minutes
        timeDiff = Math.floor(timeDiff / 60); // 63
        const minutes = Math.round(timeDiff % 60); // 3
 ```

We would compare `MAX_SCREENRECORD_LENGTH_IN_MINUTES` to 3 minutes, instead of to 63 minutes, without the suggested change.

Reply:
Feel free to verify the example above :) Didn't test out the code, just used a calculator 😅